backend/service/k8s/deployment.go

```diff
@@ -101,12 +101,144 @@ func ProtoForDeploymentSpec(deploymentSpec appsv1.DeploymentSpec) *k8sapiv1.Depl
+				LivenessProbeGRPC := &k8sapiv1.GRPCAction{
+					Port:    container.LivenessProbe.ProbeHandler.GRPC.Port,
+					Service: *container.LivenessProbe.ProbeHandler.GRPC.Service,
+				}
+				HandlerObj.Handler = &k8sapiv1.Probe_Grpc{
+					Grpc: LivenessProbeGRPC,
+				}
+			}
+
+			LivenessProbeObject = &k8sapiv1.Probe{
+				InitialDelaySeconds:           &container.LivenessProbe.InitialDelaySeconds,
+				TimeoutSeconds:                &container.LivenessProbe.TimeoutSeconds,
+				PeriodSeconds:                 &container.LivenessProbe.PeriodSeconds,
+				SuccessThreshold:              &container.LivenessProbe.SuccessThreshold,
+				FailureThreshold:              &container.LivenessProbe.FailureThreshold,
+				TerminationGracePeriodSeconds: container.LivenessProbe.TerminationGracePeriodSeconds,
+				Handler:                       HandlerObj.Handler,
+			}
+		}
+
+		if container.ReadinessProbe != nil {
+			HandlerObj := &k8sapiv1.Probe{}
+			if container.ReadinessProbe.ProbeHandler.HTTPGet != nil {
+				ReadinessProbeHTTPHeaders := make([]*k8sapiv1.HTTPHeader, 0, len(container.ReadinessProbe.ProbeHandler.HTTPGet.HTTPHeaders))
+				for _, value := range container.ReadinessProbe.ProbeHandler.HTTPGet.HTTPHeaders {
+					UniqueReadnessHeader := &k8sapiv1.HTTPHeader{
+						Name:  &value.Name,
+						Value: &value.Value,
+					}
+					ReadinessProbeHTTPHeaders = append(ReadinessProbeHTTPHeaders, UniqueReadnessHeader)
+				}
+				ReadinessProbeHTTPObject := &k8sapiv1.HTTPGetAction{
+					Path:        container.ReadinessProbe.ProbeHandler.HTTPGet.Path,
+					Port:        container.ReadinessProbe.ProbeHandler.HTTPGet.Port.IntVal,
+					Host:        container.ReadinessProbe.ProbeHandler.HTTPGet.Host,
+					Scheme:      string(container.ReadinessProbe.ProbeHandler.HTTPGet.Scheme),
+					HttpHeaders: ReadinessProbeHTTPHeaders,
+				}
+				HandlerObj.Handler = &k8sapiv1.Probe_HttpGet{
+					HttpGet: ReadinessProbeHTTPObject,
+				}
+			}
+
+			if container.ReadinessProbe.ProbeHandler.Exec != nil {
+				ReadinessProbeExec := &k8sapiv1.ExecAction{
+					Command: container.ReadinessProbe.ProbeHandler.Exec.Command,
+				}
+				HandlerObj.Handler = &k8sapiv1.Probe_Exec{
+					Exec: ReadinessProbeExec,
+				}
+			}
+
+			if container.ReadinessProbe.ProbeHandler.TCPSocket != nil {
+				ReadinessProbeTCPSocket := &k8sapiv1.TCPSocketAction{
+					Port: container.ReadinessProbe.ProbeHandler.TCPSocket.Port.IntVal,
+					Host: container.ReadinessProbe.ProbeHandler.TCPSocket.Host,
+				}
+				HandlerObj.Handler = &k8sapiv1.Probe_TcpSocket{
+					TcpSocket: ReadinessProbeTCPSocket,
+				}
+			}
+
+			if container.ReadinessProbe.ProbeHandler.GRPC != nil {
+				ReadinessProbeGRPC := &k8sapiv1.GRPCAction{
+					Port:    container.ReadinessProbe.ProbeHandler.GRPC.Port,
+					Service: *container.ReadinessProbe.ProbeHandler.GRPC.Service,
+				}
+				HandlerObj.Handler = &k8sapiv1.Probe_Grpc{
+					Grpc: ReadinessProbeGRPC,
+				}
+			}
```

Comment:
nit: maybe we should use a switch here? only one of these can be configured at a time.

Reply:
That is not possible I cannot filter by type this structure